src/middlewares/authenticator.js

```diff
@@ -35,34 +33,55 @@ module.exports = async function (req, res, next) {
 			}
 		})
 
-		common.guestUrls.map(function (path) {
-			if (req.path.includes(path)) {
-				guestUrl = true
-			}
-		})
 		common.roleValidationPaths.map(function (path) {
 			if (req.path.includes(path)) {
 				roleValidation = true
 			}
 		})
-		common.apiPermissionsUrls.map(function (path) {
-			if (req.path.includes(path)) {
-				apiPermissions = true
-			}
-		})
 
-		if ((internalAccess || guestUrl || apiPermissions) && !authHeader) {
+		if (internalAccess && !authHeader) {
 			next()
 			return
 		}
 
 		if (!authHeader) {
-			throw common.failureResponse({
-				message: 'UNAUTHORIZED_REQUEST',
-				statusCode: httpStatusCode.unauthorized,
-				responseCode: 'UNAUTHORIZED',
-			})
+			try {
+				const response = await userRequests.getListOfUserRoles(
```

Comment:
@sumanvpacewisdom use the API to find the public role, instead of fetching all the role